pandas/core/internals/construction.py

```diff
@@ -966,8 +966,9 @@ def convert(arr):
             if dtype is None:
                 if arr.dtype == np.dtype("O"):
                     # i.e. maybe_convert_objects didn't convert
-                    arr = maybe_infer_to_datetimelike(arr)
-                    if dtype_backend != "numpy" and arr.dtype == np.dtype("O"):
+                    convert_to_nullable_dtype = dtype_backend != "numpy"
+                    arr = maybe_infer_to_datetimelike(arr, convert_to_nullable_dtype)
```

Comment:
I was surprised by the fact that `maybe_infer_to_datetimelike` would change an object dtype to a string datatype; might be a cause of unexpected behavior elsewhere where this is used

Reply:
(not necessarily for now, but long term I think we should directly use `maybe_convert_objects` here instead of `maybe_infer_to_datimelike` as a small wrapper around it)

Reply:
Sounds good - I opened https://github.com/pandas-dev/pandas/issues/60258 to keep track of this